lms/djangoapps/certificates/signals.py

```diff
@@ -90,8 +96,11 @@ def fire_ungenerated_certificate_task(user, course_key):
     mode_is_verified = enrollment_mode in GeneratedCertificate.VERIFIED_CERTS_MODES
     cert = GeneratedCertificate.certificate_for_student(user, course_key)
     if mode_is_verified and (cert is None or cert.status == 'unverified'):
-        generate_certificate.apply_async(kwargs={
+        kwargs = {
             'student': unicode(user.id),
-            'course_key': unicode(course_key),
-        })
+            'course_key': unicode(course_key)
+        }
+        if expected_verification_status:
+            kwargs['expected_verification_status'] = unicode(expected_verification_status)
+        generate_certificate.apply_async(kwargs=kwargs)
```

Comment:
You'll want to add a `countdown` option to the `apply_async` method in order to delay the initial call to avoid the race condition.